### PR TITLE
Remove token validation

### PR DESCRIPTION
**Pull request by salman90**
# Pull Request

## Purpose

Removing token validation from identity-express by:
  1- Removing TokenValidator class.
  2- Removing TokenValidator class in the test.
  3- Adding decoding token to  CryptoUtils class.
* ...

## Does this introduce a breaking change

```console
    [ ] Yes
    [x] No
```

## Pull request type

What kind of change does this Pull Request introduce?
Removing token validation from 

```console
    [ ] Bugfix
    [ ] Feature
    [x] Code style update (formatting, local variables)
    [x] Documentation content changes
    [ ] Other... Please describe:
```

## How to test

* Get the code

```console
    git clone [repo-address]
    cd [repo-name]
    git checkout [branch-name]
    npm install
```

## What to check

ex: verify that the following are valid:

* ...

### CODE REVIEW

**Review by jo-arroyo — APPROVED** [review]
LGTM!

File: src/client/webapp/AppServiceWebAppAuthClient.ts [hunk 1]
```diff
@@ -41,13 +42,16 @@ import {
...
+    private cryptoUtils: CryptoUtils;
+
     /**
      * @param {AppSettings} appSettings
      * @param {Configuration} msalConfig
      * @constructor
      */
     constructor(appSettings: AppSettings, msalConfig: Configuration) {
         super(appSettings, msalConfig);
+        this.cryptoUtils = new CryptoUtils();
```
Comment (derisen) [inline src/client/webapp/AppServiceWebAppAuthClient.ts:54]:
```suggestion
        this.cryptoUtils = new CryptoUtils();
```

Can we instantiate this in `BaseAuthClient.ts` instead?

File: src/utils/CryptoUtils.ts [hunk 2]
```diff
@@ -29,4 +37,17 @@ export class CryptoUtils {
         const decipher = crypto.createDecipheriv(this.algorithm, key, Buffer.from(iv, "hex"));
         return decipher.update(encrypted, 'hex', 'utf8') + decipher.final('utf8');
     }
+
+    decodeAuthToken(authToken: string): any {
```
Comment (derisen) [inline src/utils/CryptoUtils.ts:41]:
```suggestion
    decodeAuthToken(authToken: string): any {
```

Let's specify the function return type

File: src/utils/CryptoUtils.ts [hunk 3]
```diff
@@ -29,4 +37,17 @@ export class CryptoUtils {
...
+        let base64Url = authToken.split('.')[1];
+        let base64  =  base64Url.replace(/-/g,'+').replace(/_/, '/')
+        let jsonPayload = decodeURIComponent(atob(base64).split('').map(function(c){
+                return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2)
+        }).join(''));
+        let jsonParse = JSON.parse(jsonPayload);
+        obj["payload"] = jsonParse;
+        return obj; 
+    }
+
```
Comment (derisen) [inline src/utils/CryptoUtils.ts:52]:
```suggestion
        let obj = {};
        let base64Url = authToken.split('.')[1];
        let base64  =  base64Url.replace(/-/g,'+').replace(/_/, '/')
        let jsonPayload = decodeURIComponent(atob(base64).split('').map(function(c){
                return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2)
        }).join(''));
        let jsonParse = JSON.parse(jsonPayload);
        obj["payload"] = jsonParse;
        return obj; 
    }
```

`atob`/`btoa` modules shouldn't be used on the Node.js side (it's deprecated in favour of the `Buffer` class).

In fact, we should import the decoding functionality from msal-node & msal-common directly, or duplicate that code if its not exported. What we need here is the `extractTokenClaims` method from the [AuthToken](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/src/account/AuthToken.ts) class in msal-common, using the [base64decode](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/38b7fe045288993541250f240a8df081982d357e/lib/msal-node/src/crypto/CryptoProvider.ts#L49) implementation from msal-node.

Comment (salman90) [reply src/utils/CryptoUtils.ts]: > `atob`/`btoa` modules shouldn't be used on the Node.js side (it's deprecated in favour of the `Buffer` class).
> 
> In fact, we should import the decoding functionality from msal-node & msal-common directly, or duplicate that code if its not exported. What we need here is the `extractTokenClaims` method from the [AuthToken](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/src/account/AuthToken.ts) class in msal-common, using the [base64decode](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/38b7fe045288993541250f240a8df081982d357e/lib/msal-node/src/crypto/CryptoProvider.ts#L49) implementation from msal-node.

Updated; Thanks for the feedback!

**Review by derisen — APPROVED** [review]
LGTM, thanks @salman90!

**Pull request merged**